packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,202 @@
+  /**
+   * Import an existing notification rule provided an ARN
+   * @param scope The parent creating construct
+   * @param id The construct's name
+   * @param notificationRuleArn Notification rule ARN (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   */
+  public static fromNotificationRuleArn(scope: constructs.Construct, id: string, notificationRuleArn: string): INotificationRule {
+    class Import extends Resource implements INotificationRule {
+      readonly notificationRuleArn = notificationRuleArn;
+
+      public addTarget(_target: INotificationRuleTarget): boolean {
+        return false;
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * The target config of notification rule
+   */
+  private targets: NotificationRuleTargetConfig[] = [];
+
+  /**
+   * The events of notification rule
+   */
+  private events: string[] = [];
+
+  constructor(scope: constructs.Construct, id: string, props: NotificationRuleProps) {
+    super(scope, id);
+
+    const source = props.source.bindAsNotificationRuleSource(this, this);
+
+    if (props.events) {
+      this.addEvents(props.events);
+    }
+
+    props.targets?.forEach((target) => {
+      this.addTarget(target);
+    });
+
+    const resource = new CfnNotificationRule(this, 'Resource', {
+      name: props.notificationRuleName || this.generateName(),
+      detailType: props.detailType || DetailType.FULL,
+      targets: this.targets,
+      eventTypeIds: this.events,
+      resource: source.sourceArn,
+      status: props.enabled !== undefined
+        ? (!props.enabled ? 'DISABLED' : 'ENABLED')
+        : undefined,
+    });
+
+    this.notificationRuleArn = resource.ref;
+  }
+
+  /**
+   * Adds target to notification rule
+   * @param target The SNS topic or AWS Chatbot Slack target
+   */
+  public addTarget(target: INotificationRuleTarget): boolean {
+    this.targets.push(target.bindAsNotificationRuleTarget(this, this));
+    return true;
+  }
+
+  /**
+   * Adds events to notification rule
+   *
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#events-ref-pipeline
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#events-ref-buildproject
+   * @param events The list of event types for AWS Codebuild and AWS CodePipeline
+   */
+  private addEvents(events: string[]): void {
+    events.forEach((event) => {
+      if (this.events.includes(event)) {
+        return;
+      }
+
+      this.events.push(event);
+    });
+  }
+
+  /**
+   * The name length generated by id must short than 64 characters
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-name
+   * @returns string
+   */
+  private generateName(): string {
+    return Names.uniqueId(this).slice(-64);
```

Comment:
Let's just inline this method. Doesn't seem super valuable to have it separately, it's only used in one place.

Reply:
Ok, makes sense, I will adjust.